include/s3select_functions.h

```diff
@@ -908,14 +908,21 @@ struct _fn_like : public base_function
 {
 
   value res;
+  std::regex x;
```

Comment:
please use better name (something like compiled-regex)
review comment bellow.

constructor should run transform on right expression (should be constant string in s3select query)
and populate x (actually constructing x)